downloads/swan-lake-release-notes/swan-lake-2201.0.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,359 @@
+            self.address = address.cloneReadOnly();
+      }
+      
+      function getAddress() returns string => string:'join(", ", ...self.address);
+   }
+      
+   public function main() {
+      Person person = new ("May", ["Palm Grove", "Colombo 3"]);
+      
+      // This, which was allowed previously results in an error now.
+      readonly readOnlyValue = person;
+   }
+   ```
+
+- Fixed a bug that resulted in invalid table lookups due to not distinguishing between `int`, `float`, and `decimal` zero.
+
+  ```ballerina
+  import ballerina/io;
+   
+  type Employee record {|
+     readonly int|float|decimal age;
+     int code;
+  |};
+   
+  public function main() {
+     table<Employee> key(age) empTable = table [
+         {age: 0, code: 100} // int `0` is the key
+     ];
+   
+     Employee? v1 = empTable[0f]; // Lookup with float `0` as the key.
+     io:println(v1 is Employee); // Previously `true`, now `false`.
+   
+     Employee? v2 = empTable[0d]; // Lookup with decimal `0` as the key.
+     io:println(v2 is Employee); // Previously `true`, now `false`.
+   
+     Employee? v3 = empTable[0]; // Lookup with int `0` as the key.
+     io:println(v3 is Employee); // Only this will continue to be `true`.
+  }
+  ```
+
+To view bug fixes, see the [GitHub milestone for 2201.0.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.0.0%22).
+
+### Runtime Updates
+
+#### Bug Fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.0.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FjBallerina+milestone%3A%22Ballerina+2201.0.0%22).
+
+### Standard Library Updates
+
+#### New Features
+
+##### `graphql` Package
+- Added the CORS configuration support
+
+##### `http` Package
+
+- Implemented typed `headers` for the HTTP response
+- Added the `map<string>` data binding support for `application/www-x-form-urlencoded`
+- Added support to provide an inline request/response body with `x-form-urlencoded` content
+- Added compiler plugin validation for usage of the `@http:Payload` annotation
+
+#### Improvements
+
+##### `graphql` Package
+- Removed the deprecated `add` method in the `graphql:Context` object.
+
+##### `grpc` Package
+- Changed the `--proto_path` option of the gRPC command to `--proto-path`
+
+##### `websub` Package
+- Added support for `readonly` parameters in remote methods
+
+#### `websubhub` Package
+- Added support for `readonly` parameters in remote methods
+
+##### `kafka` Package
+- Made the `kafka:Caller` optional in the `onConsumerRecord` method of the `kafka:Service`
+- Allowed the `readonly & kafka:ConsumerRecord[]` parameter type in the `onConsumerRecord` method
+
+### Code to Cloud Updates
+
+#### Improvements
+- Removed the `awslambda` and `azure_functions` packages support in single file projects
```

Comment:
```suggestion
- Removed `awslambda` and `azure_functions` module support for single `.bal` files
```

Same as https://github.com/ballerina-platform/ballerina-dev-website/pull/3669#discussion_r795710310

Reply:
@anuruddhal

Reply:
+1. lets commit the suggestion